src/ListResult.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+namespace SynergiTech\Creditsafe;
+
+/*
+    This class is used to help paginate the results of endpoints
+*/
+class ListResult implements \Iterator
+{
+    private $client;
+    private $targetClass;
+    private $endpoint;
+    private $params;
+    private $pageSize = 10;
+    private $currentPagePos = null;
+    private $currentRecordPos = null;
+    private $currentRecordSet = [];
+
+    /**
+     * This constructor is used to build the ListResult
+     * @param Client $client       Used to set the client in the ListResult Class
+     * @param string $targetClass  Used to set a dynamic class in the LisrResult Class
+     *  allowing for ListResult to be used on different classes
+     * @param string $endpoint    Used to set the endpoint for which the ListResult Class is using
+     * @param array $params      Used tp set the params for which the ListResult Class is using
+     */
+    public function __construct(Client $client, string $targetClass, string $endpoint, array $params)
+    {
+        $this->client = $client;
+        $this->targetClass = $targetClass;
+        $this->endpoint = $endpoint;
+        $this->params = $params;
+        $this->page(1);
+    }
+
+    /**
+     * Set PageSize
+     * @param  int  $size Stores the PageSize
+     * @return self   Returns PageSize
```

Comment:
```suggestion
     * @return self
```
wrong description, probably doesn't need one